apps/meteor/app/apps/server/converters/rooms.js

```diff
@@ -38,11 +38,14 @@ export class AppRoomsConverter {
 		let v;
 		if (room.visitor) {
 			const visitor = await LivechatVisitors.findOneById(room.visitor.id);
+
+			const lastMessageTs = room?.visitor?.lastMessageTs;
 			v = {
 				_id: visitor._id,
 				username: visitor.username,
 				token: visitor.token,
 				status: visitor.status || 'online',
+				...(lastMessageTs && { lastMessageTs }),
 			};
 		}
```

Comment:
why not?
```suggestion
			v = {
				...room?.visitor,
				_id: visitor._id,
				username: visitor.username,
				token: visitor.token,
				status: visitor.status || 'online',
			};
		}
```

Reply:
I guess this could be a problem if there are properties that have been renamed when sent over to the Apps-Engine. For instance, this would make both `id` and `_id` properties available in the converted `v` variable

Reply:
What I think would work is spreading the `visitor` instead of `rooms?.visitor`

Reply:
ok, maybe my question was not clear enough, I suggested "spread" the visitor not because I was not aware that both types are different, but was more like "are we not missing any other field? or in the feature if we add another field the bridge will remove it"?  and actually looks like we are, the phone field, but thankfully I believe this field doesnt even exist so we are fine

Reply:
I get your point @ggazzo but I can't think of a way to apply your suggestions here. You see right now, app's and core are fundamentally storing 2 different objects which at first glance seems identical but they aren't. 
- The `room.v` prop on core is supposed to store information related to visitor + **ITS CURRENT CHANNEL**. So suppose you have a contact whom we're contacting using SMS via 2 phone no's, let's call X and Y. Then if the contact sends a message using X phone number, then `room.v.phoneNo` would be `X` and correspondingly we'll store the timestamp of the last message from this visitor from X phone no on `room.v.lastMessageTs`. 
- Now on the app's side, the bridge is mapping `room.v` to `room.visitor` and this visitor object only contains visitor info and **DOESN'T CONTAIN ANY INFO ABOUT THE CHANNEL**. 

This PR aims to add the additional channel info to "room.visitor" level on the apps. Right now we have 2 such channel props, one is lastMessageTs and other is phone (thanks for pointing this out btw, I'll update this PR with it shortly ;) ).
Now coming back to your suggestion, as pointed out by Kevin, room.visitor on app's has a bunch of more visitor related props which we're not storing on room.v level on core. Due to this we're manually mapping these values here. I'll further add comments to show this however if you can think of a better way to manage these typings, please feel free to share.